cfme/ansible/repositories.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+"""Page model for Automation/Anisble/Repositories"""
+from navmazing import NavigateToAttribute, NavigateToSibling
+
+from cfme.base import Server
+from cfme.exceptions import ItemNotFound
+from cfme.base.login import BaseLoggedInPage
+from widgetastic.widget import Text, Checkbox
+from widgetastic_manageiq import Table
+from widgetastic_patternfly import Dropdown, Button, Input, FlashMessages
+from utils.appliance import Navigatable
+from utils.appliance.implementations.ui import navigator, navigate_to, CFMENavigateStep
+from utils.wait import wait_for
+
+
+class RepositoryBaseView(BaseLoggedInPage):
+    flash = FlashMessages('.//div[starts-with(@class, "flash_text_div") or @id="flash_text_div"]')
+    title = Text(locator=".//h1")
+
+    @property
+    def in_ansible_repositories(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ["Automation", "Ansible", "Repositories"]
+        )
+
+
+class RepositoryListView(RepositoryBaseView):
+    configuration = Dropdown("Configuration")
+    repositories = Table(".//div[@id='list_grid']/table")
+
+    @property
+    def is_displayed(self):
+        return self.in_ansible_repositories and self.title.text == "Repositories"
+
+
+class RepositoryDetailsView(RepositoryBaseView):
+    configuration = Dropdown("Configuration")
+    download = Button(title="Download summary in PDF format")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_ansible_repositories and
+            self.title.text == "{} (Summary)".format(self.context["object"].name)
+        )
+
+
+class RepositoryFormView(RepositoryBaseView):
+    name = Input(name="name")
+    description = Input(name="description")
+    url = Input(name="scm_url")
+    scm_credentials = Dropdown("Select credentials")
+    scm_branch = Input(name="scm_branch")
+    # SCM Update Options
+    clean = Checkbox(name="clean")
+    delete_on_update = Checkbox(name="scm_delete_on_update")
+    update_on_launch = Checkbox(name="scm_update_on_launch")
+
+    cancel_button = Button("Cancel")
+
+
+class RepositoryAddView(RepositoryFormView):
+    add_button = Button("Add")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_ansible_repositories and
+            self.title.text == "Add new Repository"
+        )
+
+
+class RepositoryEditView(RepositoryFormView):
+    save_button = Button("Save")
+    reset_button = Button("Reset")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_ansible_repositories and
+            self.title.text == 'Edit Repository "{}"'.format(self.context["object"].name)
+        )
+
+
+class RepositoryCollection(Navigatable):
+    """Collection object for the :py:class:`Repository`."""
+
+    def create(self, name, url, description=None, scm_credentials=None, scm_branch=None,
```

Comment:
@quarckster Though I think we could make all parameters optional to enable us testing that it is indeed required by the UI :)